erpnext/accounts/doctype/payment_entry/payment_entry_list.js

```diff
@@ -1,12 +1,52 @@
 frappe.listview_settings['Payment Entry'] = {
 
-	onload: function(listview) {
-		listview.page.fields_dict.party_type.get_query = function() {
+	onload: function (listview) {
+		listview.page.fields_dict.party_type.get_query = function () {
 			return {
 				"filters": {
 					"name": ["in", Object.keys(frappe.boot.party_account_types)],
 				}
 			};
 		};
+		listview.page.add_actions_menu_item(__('Print Check'), () => {
+			const selected_docs = listview.get_checked_items();
+			const doctype = listview.doctype;
+			if (selected_docs.length > 0) {
+				for (let doc of selected_docs) {
+					if (doc.docstatus !== 0) {
```

Comment:
Are you sure this should only happen for Draft documents? Because draft documents can be changed, which means any printed checks can be wrong.

Reply:
Yes i have confirmed with Arundhati about this